sklearn/linear_model/stochastic_gradient.py

```diff
@@ -724,6 +724,14 @@ class SGDClassifier(BaseSGDClassifier):
         When set to True, reuse the solution of the previous call to fit as
         initialization, otherwise, just erase the previous solution.
 
+        Repeatedly calling fit or partial_fit when warm_start is True can
+        result in a different solution than when calling fit a single time
+        because of the way the data is shuffled.
+        If a dynamic learning rate is used, the learning rate is adapted
+        depending on the number of samples already seen. Calling `fit` resets
+        this counter, while `partial_fit` will result in the expected
```

Comment:
Should we clarify "expected behavior"? "while calling ``partial_fit`` will result in consecutive calls increasing the existing counter" or something like that. I find "expected behavior" a bit vague ;)